src/FrameDecorator/Page.php

```diff
@@ -357,14 +357,33 @@ protected function _page_break_allowed(Frame $frame)
 
                 // Rule C
                 $block_parent = $frame->find_block_parent();
-                if (count($block_parent->get_line_boxes()) < $frame->get_style()->orphans) {
+                $block_line_boxes = $block_parent->get_line_boxes();
+                $frame_line_number = 0;
```

Comment:
I think this should be initialized as `$frame_line_number = count($block_line_boxes)`, basically as before.
Reason being that the frame might not have been added to the line boxes yet, because we are currently in the process of reflowing its children. This can definitely happen in the case of inline-block frames (which would become a concern with #2497, when line breaks before inline-block frames are enabled).

Reply:
I think this makes sense. The line count will always be the current number of rendered lines. And if Dompdf has to walk back through previous siblings it should be able to get a valid line number.